### DISCUSSION TITLE
How to ensure that prisma will use a same connection for a series of query?

### DISCUSSION THREAD

**Discussion by luatnd** [prisma/prisma#21580] [Q&A] [answered]
### Question

Hi, I just have a question about how prisma handle connection.

As I can find in the docs, prisma have a connection pool, my nodejs application  (with DEBUG) also said that prisma have established a connection pool with 17 connections.

As I guess, if I use `prisma.$transaction` then all query & command inside will use a same connection right?

My second question is the following code will have chance to use how many connections?
```
await this.prisma.$executeRawUnsafe(op === TxOperation.Begin ? 'BEGIN' : op === TxOperation.Commit ? 'COMMIT' : 'ROLLBACK');

const a: any[] = await this.prisma.$queryRawUnsafe(
    `
      SELECT * FROM inv__game_item_inventory
      WHERE id = $1 LIMIT 1
      FOR UPDATE
    `,
    item_id,
);
```

**Comment by janpio** [comment]
> As I guess, if I use `prisma.$transaction` then all query & command inside will use a same connection right?

Correct.



> My second question is the following code will have chance to use how many connections?

Can you reformulate your question? I am not sure I understand.

**Reply by luatnd** [reply]
Ah Thank janpio, In the 2nd question, I mean my code snippet above can use up to 2 connections right?

**Reply by ludralph** [reply, marked as answer]
Hi @luatnd 

1. The `await this.prisma.$executeRawUnsafe(...)` call will reserve a connection from the pool to execute the transaction command (BEGIN, COMMIT, or ROLLBACK).

2. The `await this.prisma.$queryRawUnsafe(...)` call will also reserve a connection from the pool to execute the `SELECT` query.

If these two calls are not wrapped within a `prisma.$transaction` block, they could potentially use two separate connections from the pool, one for each call. However, if they are part of the same transaction block, they would use the same connection.
